packages/thirdweb/src/react/core/hooks/connection/useAutoConnect.tsx

```diff
@@ -193,6 +206,9 @@ export function AutoConnect(props: AutoConnectProps) {
           });
 
           connect(activeWallet);
```

Comment:
Should we wait till after `connect` has resolved before calling `onConnect`?

Reply:
Good point - await-ing is now necessary because `connect` is not simply setting is active ( which was the case earlier ) - but now it may or may not connect a smart wallet instead, so need to await it and pass the returned wallet to `onConnect` - Good catch !

Reply:
Done